ophidian/ophidian

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env sh
+
+set -eu
+
+# This script involves just enough shell to bootstrap our way to a working
+# system where we don't have to write any more shell and everything can
+# be in Python. Its responsibilities are getting us a working version of
+# pyenv, getting us a working version of Python, and then getting out of
+# the way and running Python.
+
+# Write stdout to stderr, preserving the original stdout in fd 3
+exec 3<&1 1>&2
+
+ROOT="$(dirname $0)"
+REAL_MAIN="$ROOT/src/ophidian/__main__.py"
+
+
+if [ ! -e $REAL_MAIN ] ; then
+    echo "ophidian script is in the wrong directory (expected $REAL_MAIN to exist)" >&2
+    exit 1
+fi
+
+export OPHIDIAN_HOME="${OPHIDIAN_HOME:-"$HOME/.cache/ophidian-storage"}"
+
+localtmp () {
+  # Hack to get mktemp working on Darwin properly.
+  mkdir -p $1
+  OUR_TMP_DIR=$(TMPDIR=$1 mktemp -d 2>/dev/null || TMPDIR=$1 mktemp -d -t 'clonedir')
+}
+
+# First look for pyenv. If there's one installed on the system
+# we'll use that. Otherwise we'll clone our own.
+
+OPHIDIAN_PYENV="$HOME/.pyenv"
+if [ ! -e "$OPHIDIAN_PYENV" ] ; then
+  OPHIDIAN_PYENV="$OPHIDIAN_HOME/.pyenv"
+fi
+
+export OPHIDIAN_PYENV
+
+if [ ! -e "$OPHIDIAN_PYENV" ] ; then
+  localtmp "$OPHIDIAN_HOME/tmp"
+  git clone https://github.com/yyuu/pyenv.git $OUR_TMP_DIR/pyenv;
```

Comment:
nit: this doesn't specify any versioning info

I don't know how likely that is to cause issues, but it's worth mentioning